test/services/ProveGateway/reactTo.test.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  Gateway,
+  GatewayRepository,
+} from '../../../src/models/GatewayRepository';
+import { MessageRepository } from '../../../src/models/MessageRepository';
+
+import chai = require('chai');
+import chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+const { assert } = chai;
+
+const Mosaic = require('@openst/mosaic.js');
+const Web3 = require('web3');
+
+describe('ProveGatewayService.reactTo()', () => {
+  const originWeb3 = new Web3();
+  const auxiliaryWeb3 = new Web3();
+  const auxiliaryWorkerAddress = '0xF1e701FbE4288a38FfFEa3084C826B810c5d5294';
+  const gatewayAddress = '0x0000000000000000000000000000000000000001';
+  const blockNumber = new BigNumber(2);
+
+  it('should react to block height of new anchor state root', async () => {
+    const gatewayRecord: Gateway = StubData.gatewayRecord();
+    const gateawayRepository = sinon.createStubInstance(GatewayRepository, {
+      get: Promise.resolve(gatewayRecord),
+    });
+
+    const messageRepository = sinon.createStubInstance(MessageRepository, {
+      isPendingMessages: Promise.resolve(true),
+    });
+
+    const proof = { encodedAccountValue: 'encodedAccountValue', serializedAccountProof: 'serializedAccountProof' };
+    const proofGeneratorStub = sinon.replace(
+      Mosaic.Utils.ProofGenerator.prototype,
+      'getOutboxProof',
+      sinon.fake.resolves(proof),
+    );
+
+    const fakeReceipt = { status: true };
+    const coGatewayStub = sinon.replace(
+      Mosaic.ContractInteract.EIP20CoGateway.prototype,
+      'proveGateway',
+      sinon.fake.resolves(fakeReceipt),
+    );
+    const proveGatewayService = new ProveGatewayService(
+      gateawayRepository as any,
+      messageRepository as any,
+      originWeb3,
+      auxiliaryWeb3,
+      auxiliaryWorkerAddress,
+      gatewayAddress,
+    );
+
+    const response = await proveGatewayService.reactTo(blockNumber);
+
+    SpyAssert.assert(
+      gateawayRepository.get,
+      1,
+      [[gatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      messageRepository.isPendingMessages,
+      1,
+      [[blockNumber, gatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      proofGeneratorStub,
+      1,
+      [[gatewayRecord.gatewayAddress, [], blockNumber]],
+    );
+
+    SpyAssert.assert(
+      coGatewayStub,
+      1,
+      [[
+        blockNumber,
+        proof.encodedAccountValue,
+        proof.serializedAccountProof,
+        { from: auxiliaryWorkerAddress }]],
+    );
+
+    assert.strictEqual(
+      fakeReceipt,
+      response.receipt,
+      'Service must return expected receipt',
+    );
+
+    assert.strictEqual(
+      response.success,
+      true,
+      'Success must be true',
+    );
+    assert.strictEqual(
+      response.message,
+      'Gateway successfully proven',
+      'Service must return correct messages',
+    );
+  });
+
+  it('should fail to react if gayeway details does not exists', async (): Promise<void> => {
```

Comment:
`it('should fail to react if gateway details does not exists', async (): Promise<void> => {`